### PR TITLE
Improved TRD digitizer (O2-455)

### PR DESCRIPTION
**Pull request by jolopezl**
- More and better hardcoded values (waiting for ccdb)
- Unnecesary messages removed
- Almost finalized conversion of hits to signals (missing signal vector filling)

O2-455

### CODE REVIEW

File: Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h [hunk 1]
```diff
@@ -11,7 +11,13 @@
 #ifndef ALICEO2_TRD_DIGITIZER_H_
 #define ALICEO2_TRD_DIGITIZER_H_
 
+#include "TRDBase/TRDGeometry.h"
+#include "TRDBase/TRDCommonParam.h" // For kNdet & el. diffusion
+#include "TRDBase/TRDSimParam.h"
 #include "TRDBase/Digit.h"
+#include "TRDBase/TRDArraySignal.h"
+#include "TRDBase/PadResponse.h"
+
```
Comment (davidrohr) [inline Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h]: Do you need all these includes? Can't you forward declare them.
As far as I can see, they are not used in the header file.

**Comment by davidrohr** [comment]
@jolopezl : The C.I. problem seems to be fixed. Can you have a look whether the includes are needed? Then we can merge this.

File: Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h (hunk 1 above)
Comment (jolopezl) [reply Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h]: @davidrohr, I am using forward declarations now. Thanks for the advice.

**Comment by davidrohr** [comment]
@jolopezl : Thx, looks better, can you fix the clang-format. If it passes the C.I. then, we can merge.

**Comment by jolopezl** [comment]
@davidrohr, clang-format fixed.

**Comment by davidrohr** [comment]
@dberzano @ktf : The CI is again failing due to memory issues: "sh: fork: Cannot allocate memory"
Is someone working on this?

**Comment by jolopezl** [comment]
Fails in o2checkcode/o2 tests look awful, but apparently unrelated to this PR.

**Comment by jolopezl** [comment]
@davidrohr, all tests passing now.

**Pull request merged**